### PR TITLE
HEEDLS-562 Add cancel link component and replace back links with it

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-562

### Description
Added a new Cancel Link view component that is the same as the back link, but has a cross instead of the <. Replaced all instances of the Back link component with text Cancel with this component. Also replaced some "< Go back" back links in multi page form journeys with Back buttons.

### Screenshots
A few examples. Styling should be the same across all of them
image
image
image

image
image

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/RegisterDelegateByCentre/Summary.cshtml [hunk 1]
```diff
@@ -134,6 +134,6 @@
     <form method="post" asp-action="Summary">
       <button class="nhsuk-button" type="submit">Save</button>
     </form>
-    <vc:back-link asp-controller="RegisterDelegateByCentre" asp-action="@Model.PreviousAction" link-text="Go back" />
+    <vc:cancel-link asp-controller="AllDelegates" asp-action="Index" />
```
Comment (ibrahimmunir14) [inline DigitalLearningSolutions.Web/Views/RegisterDelegateByCentre/Summary.cshtml:137]: I think this was the only use of PreviousAction, could you remove it from the VM please if there are no other references to it.

File: DigitalLearningSolutions.Web/Styles/shared/components/cancelLink.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+.nhsuk-back-link__link {
+  padding-left: 24px;
+}
+
+.nhsuk-icon__close {
+  height: 24px;
+  left: -2px; /* 2 */
+  position: absolute;
+  top: -1px; /* 3 */
```
Comment (ibrahimmunir14) [inline DigitalLearningSolutions.Web/Styles/shared/components/cancelLink.scss:11]: What do the comments mean?

**Review by ibrahimmunir14 — CHANGES_REQUESTED** [review]
LGTM. Just two minor comments.

File: DigitalLearningSolutions.Web/Styles/shared/components/cancelLink.scss (hunk 2 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Styles/shared/components/cancelLink.scss]: I'd just copied the style from the NHS css for the back link, it has a few comments to describe what they do (basically alignment of the svg)

**Review by ibrahimmunir14 — APPROVED** [review]

**Review by livzorn — APPROVED** [review]
Looks good to me!

**Pull request merged**